### ISSUE TITLE
[EAM] [CAM] [ifx] [OpenMP] During compilation of components/eam/src/physics/cam/debug_info.F90 with ifx, build error when using OpenMP and optimization >= O1

### ISSUE THREAD

**Issue by omarkahmed** [E3SM-Project/E3SM#5522]
Compilation errors with components/eam/src/physics/cam/debug_info.F90 on JLSE due to ifx optimizations.

Simplified reproducer ("debug_info_repro.F90"): 
```
module debug_info
​
  implicit none
  private
​
  !Variables updated during the simulation
  integer :: icol_p3
  !$OMP THREADPRIVATE(icol_p3 )
​
  public:: &
       report_error_info        !report info on the current lat/lon and other simulation parameters
​
contains
​
  subroutine report_error_info(icol)
​
​
    implicit none
​
    !intent-in and optional
    integer, optional, intent(in) :: icol !column(lat/lon) index
​
    !local variables
​
    integer                :: col_to_prnt
    character(len=1000)    :: icolstr
​
    if(present(icol)) then
       icolstr     = '*NOTE*: Using user specified column index for computing lat-lons'
       col_to_prnt = icol
    else
       icolstr = '*NOTE*: Using auto updated column index from P3 for computing lat-lons'
       col_to_prnt = icol_p3
    endif
​
    write(*,*) icolstr
    write(*,*) col_to_prnt
​
  end subroutine report_error_info
​
​
end module debug_info
```

Run:
```
module use /soft/restricted/CNDA/modules
module load oneapi/eng-compiler/2022.12.30.003
Either:
- ifx -fopenmp -O1 -c ./debug_info_repro.F90 -o debug_info.F90.o
- ifx -fopenmp -O2 -c ./debug_info_repro.F90 -o debug_info.F90.o
- ifx -fopenmp -O3 -c ./debug_info_repro.F90  -o debug_info.F90.o
```

Error message will look like this:

```

          #0 0x0000000001f88142
          #1 0x0000000001feb9c7
          #2 0x0000000001febaf0
          #3 0x000014e257868090
          #4 0x0000000003f25694
          #5 0x0000000003e83bbd
          #6 0x0000000003e86661
          #7 0x0000000003e85792
          #8 0x0000000002ed759d
          #9 0x00000000023134fd
         #10 0x0000000002ecf0ed
         #11 0x0000000002b7f2f6
         #12 0x0000000002ee611d
         #13 0x0000000002b7ae34
         #14 0x0000000002ecec1d
         #15 0x0000000002b7d493
         #16 0x0000000002ee637d
         #17 0x0000000002b7c505
         #18 0x0000000002ecedfd
         #19 0x00000000023120f0
         #20 0x0000000002f3d64d
         #21 0x0000000002ed1d2d
         #22 0x00000000023120f0
         #23 0x0000000001f2d217
         #24 0x0000000001eda247
         #25 0x00000000020a0343
         #26 0x000014e257849083 __libc_start_main + 243
         #27 0x0000000001d163a9/tmp/ifx00636489170OSivt/ifxVzlbus.i90: error #5633: **Internal compiler error: segmentation violation signal raised** Please report this error along with the circumstances in which it occurred in a Software Problem Report.  Note: File and line given may not be explicit cause of this error.
```
This issue is relevant for the latest Intel compiler nightly dated 3/10/23.  Note that this is not an issue for ifort.
Current Intel JIRA: https://jira.devtools.intel.com/browse/CMPLRLLVM-45498

There is a fix at: https://github.com/omarkahmed/E3SM/commit/c4486afda457b70c2936765f7f145115b6dd8266

**Comment by omarkahmed** [comment]
@amametjanov , the create_newcase command I used is as follow:

```
E3SM=${HOME}/git-repos/E3SM-omar.revert

RES=ne4pg2_ne4pg2
MACH=jlse
COMPSET=F2010-MMF1
COMPILER=oneapi-ifxgpu
MPI_RANKS=2
NTHREADS=36
MPI_LIB=mpich

PROJ=cli115
OUTPUT=${PWD}
export CASE=${COMPSET}_${MACH}_${HARDWARE}_${MPI_LIB}_${RES}_${COMPILER}_${MPI_RANKS}x${NTHREADS}


${E3SM}/cime/scripts/create_newcase -case ${CASE} -compset ${COMPSET} -res ${RES} -mach ${MACH} -mpilib ${MPI_LIB} -compiler ${COMPILER} -project ${PROJ} --output-root ${OUTPUT} -pecount ${MPI_RANKS}x${NTHREADS} --handle-preexisting-dirs r
```

**Comment by xyuan** [comment]
This issue has been reported as a bug to Intel ifx compiler team
https://wiki.jlse.anl.gov/pages/viewpage.action?pageId=38797463

**Comment by omarkahmed** [comment]
@amametjanov , a fix has been implemented in Intel's nightly compiler (xmain branch) dated 3/16/23.

**Comment by omarkahmed** [comment]
I confirm this is resolved with ANL module oneapi/eng-compiler/2022.12.30.005, closing this out.
